rclpy/rclpy/node.py

```diff
@@ -126,6 +137,15 @@ def destroy_client(self, client):
                 return True
         return False
 
+    def destroy_timer(self, timer):
+        for tmr in self.timers:
+            if tmr.timer_handle == timer.timer_handle:
+                _rclpy.rclpy_destroy_entity(
+                    'timer', tmr.timer_handle)
```

Comment:
All the C functions are raising `RuntimeError` if anything goes wrong at the `rcl` level, and if the argument parsing fails the C-python API raises the [appropriate exception](https://docs.python.org/3/c-api/arg.html#c.PyArg_ParseTuple) for you. So the return value is not really meaningful in that context.

But I agree that error handling and asserts in this library need to be cleaned up/homogenized and it will be done in a follow-up PR.